### ISSUE TITLE
translucent png images for markers has a dark tint

### ISSUE THREAD

**Issue by zainnadeem** [tangrams/tangram-es#1782]
**TO REPRODUCE THE ISSUE, FOLLOW THESE STEPS:**

 I am trying to add a png image that has an opacity of %20 percent. 

**RESULT:**

The png is coming out dark. It seems anytime you add an image to a marker with an alpha other than 1, it darkens the image. 

![img_7708](https://user-images.githubusercontent.com/16995739/37925941-ef39d006-3103-11e8-80e7-dd4f09caf78f.jpg)

**EXPECTED RESULT:**

I expect the image to have a fully transparent background and should not tint my png. It should look like this: 

![screen shot 2018-03-26 at 10 45 43 am](https://user-images.githubusercontent.com/16995739/37913539-17db3ba6-30e3-11e8-993a-05024a4ebc61.png)

**ENVIRONMENT:**

 - Xcode 9.3 swift

**Comment by matteblair** [comment]
Hi @zainnadeem, thanks for filing an issue! I'm trying to understand the situation here. From your screenshots it looks like there is one image with a grid of dots and one image that looks like a round pink button - is that right? Are you able to share these images for us to investigate the issue?

**Comment by zainnadeem** [comment]
Hi, thanks for the quick response. I've attached the image below. It's pink image with dots in the center. When I add it as a marker to the map, it renders dark, like the first image above. 


![radius](https://user-images.githubusercontent.com/16995739/37925390-64e0b722-3102-11e8-818b-e34f674a674f.png)

**Comment by zainnadeem** [comment]
Thanks @matteblair  let me know if I can provide anything else that might be useful.

**Comment by matteblair** [comment]
With some help, I think I've found the source of this problem (thanks @karimnaaji!). When we set an icon for a map marker, we go through a series of steps to get the raw pixel data from the `UIImage` so that it can be passed into the cross-platform C++ core. In that process we create a `CGBitmapContext` and render the `UIImage` into it with a set of options, one of which is the alpha info:
https://github.com/tangrams/tangram-es/blob/master/platforms/ios/src/TangramMap/TGMarker.mm#L158

The pixel data that we want to get out is 32bpp RGBA, _not_ alpha-premultiplied. This is represented by the `kImageAlphaLast` entry in the [`CGImageAlphaInfo` enum](https://developer.apple.com/documentation/coregraphics/cgimagealphainfo?language=objc) but according to the [Quartz2D programming guide](https://developer.apple.com/library/content/documentation/GraphicsImaging/Conceptual/drawingwithquartz2d/dq_context/dq_context.html#//apple_ref/doc/uid/TP30001066-CH203-BCIBHHBB) this option is not supported on iOS (I tried it anyway and sure enough, it generates an error and fails on iOS).

Since iOS only supports alpha-premultiplied `CGBitmapContext` configurations, we will have to either:
 - un-multiply the alpha from the resulting pixels by looping over the entire image manually, or
 - find a way to extract pixel data from a `UIImage` that doesn't use `CGBitmapContext`.

I'll update when I've got a working fix!
